### DISCUSSION TITLE
what is the purpose of Traverse, please?

### DISCUSSION THREAD

**Discussion by wey-gu** [vesoft-inc/nebula#4228] [Q&A] [answered]
May I ask why we introduced it in https://github.com/vesoft-inc/nebula/pull/3308, please? How is it differentiated from getNeighbors, please?

Some background/motivation shared will be appreciated :).

**update**: I realize it's part of the multiple pattern support?

cc @CPWstatic

**Comment by jievince** [comment, marked as answer]
`GetNeighbors` is a one-hop expand,
and `Traverse` is a multi-hop expand.
Indeed, the implementation of `Traverse` uses `GetNeighbors` to do the expansion of every hop.

> update: I realize it's part of the multiple pattern support?

Yes, `Traverse` is easy to do the pattern like `-[:like*1..3]`

**Reply by wey-gu** [reply]
Thanks a lot @jievince ❤
